packages/location-field/src/index.js

```diff
@@ -0,0 +1,831 @@
+    // effective way to handle this.
+    const { location } = this.props;
+    if (location !== prevProps.location) {
+      /* FIXME only disabled this because it'd take longer to refactor */
+      /* eslint-disable-next-line */
+      this.setState({
+        value: location !== null ? location.name : "",
+        geocodedFeatures: []
+      });
+    }
+  }
+
+  getFormControlClassname() {
+    const { locationType } = this.props;
+    return `${locationType}-form-control`;
+  }
+
+  setLocation(location) {
+    const { onLocationSelected, locationType } = this.props;
+    onLocationSelected({ locationType, location });
+    this.setState({ menuVisible: false });
+  }
+
+  useCurrentLocation = () => {
+    const {
+      currentPosition,
+      getCurrentPosition,
+      onLocationSelected,
+      locationType
+    } = this.props;
+    const location = currentPositionToLocation(currentPosition);
+    if (location) {
+      // If geolocation is successful (i.e., user has granted app geolocation
+      // permission and coords exist), set location.
+      onLocationSelected({ locationType, location });
+    } else {
+      // Call geolocation.getCurrentPosition and set as from/to locationType
+      getCurrentPosition(locationType);
+    }
+    this.setState({ menuVisible: false });
+  };
+
+  /**
+   * Provide alert to user with reason for geolocation error
+   */
+  geolocationAlert = () => {
+    const { currentPosition } = this.props;
+    window.alert(
+      `Geolocation either has been disabled for ${
+        window.location.host
+      } or is not available in your browser.\n\nReason: ${currentPosition.error
+        .message || "Unknown reason"}`
+    );
+  };
+
+  onClearButtonClick = () => {
+    const { clearLocation, locationType } = this.props;
+    clearLocation({ locationType });
+    this.setState({
+      value: "",
+      geocodedFeatures: []
+    });
+    /* eslint-disable-next-line */
+    ReactDOM.findDOMNode(this.inputRef).focus();
+    this.onTextInputClick();
+  };
+
+  onDropdownToggle = () => {
+    const { menuVisible } = this.state;
+    this.setState({ menuVisible: !menuVisible });
+  };
+
+  /**
+   * Only hide menu if the target clicked is not a menu item in the dropdown.
+   * Otherwise, the click will not "finish" and the menu will hide without the
+   * user having made a selection.
+   */
+  onBlurFormGroup = e => {
+    const { location } = this.props;
+    // IE does not use relatedTarget, so this check handles cross-browser support.
+    // see https://stackoverflow.com/a/49325196/915811
+    const target =
+      e.relatedTarget !== null ? e.relatedTarget : document.activeElement;
+    if (!location && (!target || target.getAttribute("role") !== "menuitem")) {
+      this.setState({ menuVisible: false, value: "", geocodedFeatures: [] });
+    }
+  };
+
+  onTextInputChange = evt => {
+    this.setState({ value: evt.target.value, menuVisible: true });
+    this.geocodeAutocomplete(evt.target.value);
```

Comment:
We should explicitly ask which of these low-level text input listeners AltSource needs access to.

Reply:
We can ask. Or they can tell us. Or they can make a PR. :)